cmd/booster-bitswap/filters/filters.go

```diff
@@ -93,12 +93,21 @@ func (f *filter) update() error {
 		return nil
 	}
 	defer stream.Close()
-	// open the cache file
-	cache, err := os.OpenFile(f.CacheFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	// write cache to the temp file first
+	tmpFile := f.CacheFile + ".tmp"
+	cache, err := os.OpenFile(tmpFile, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
 	if err != nil {
 		return fmt.Errorf("opening cache file: %w", err)
 	}
-	defer cache.Close()
+	defer func() {
+		cache.Close()
```

Comment:
Should we check the return value of cache.Close() also?

Reply:
Great spot, fixed.